core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -173,6 +173,12 @@ protected RelBuilder(Context context, RelOptCluster cluster,
         new RexSimplify(cluster.getRexBuilder(), predicates, executor);
   }
 
+  private RelBuilder(Context context, RelOptCluster cluster,
+                     RelOptSchema relOptSchema, Deque<Frame> stack) {
+    this(context, cluster, relOptSchema);
+    this.stack.addAll(stack);
+  }
```

Comment:
This constructor is not necessary.

Reply:
this is used by withRelFactories() in order to pass in stack.

Reply:
I guess @danny0405 means the body can be just moved to `withRelFactories()`.

Reply:
that makes sense. I updated. thx.